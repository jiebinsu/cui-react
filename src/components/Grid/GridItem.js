import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";

const GridItem = ({
  className,
  size,
  offset,
  sm,
  smOffset,
  md,
  mdOffset,
  lg,
  lgOffset,
  xlg,
  xlgOffset,
  collapse,
  children,
  ...props
}) => {
  const classes = classNames("coop-l-grid__item", [
    size && `coop-l-grid__item--${size}`,
    offset && `coop-l-grid__item--offset-${offset}`,
    sm && `coop-l-grid__item--small-${sm}`,
    smOffset && `coop-l-grid__item--small-${smOffset}`,
    md && `coop-l-grid__item--medium-${md}`,
    mdOffset && `coop-l-grid__item--medium-offset-${mdOffset}`,
    lg && `coop-l-grid__item--large-${lg}`,
    lgOffset && `coop-l-grid__item--large-offset-${lgOffset}`,
    xlg && `coop-l-grid__item--xlarge-${xlg}`,
    xlgOffset && `coop-l-grid__item--xlarge-offset-${xlgOffset}`,
    collapse && "coop-l-grid__item--collapse",
    className,
  ]);

  return (
    <div className={classes} {...props}>
      {children}
    </div>
  );
};

GridItem.defaultProps = {
  className: "",
  size: null,
  offset: null,
  sm: null,
  smOffset: null,
  md: null,
  mdOffset: null,
  lg: null,
  lgOffset: null,
  xlg: null,
  xlgOffset: null,
  collapse: false,
  children: null,
};

GridItem.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  offset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  sm: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  smOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  md: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  mdOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  lg: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  lgOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  xlg: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  xlgOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  collapse: PropTypes.bool,
  children: PropTypes.node,
};

export default GridItem;
