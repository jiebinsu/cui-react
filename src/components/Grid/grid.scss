@import "@coopdigital/foundations-grid/dist/grid.css";
