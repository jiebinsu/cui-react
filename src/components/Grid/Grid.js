import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";
import "./grid.scss";

const Grid = ({ className, center, reverse, children, ...props }) => (
  <div
    className={classNames("coop-l-grid", [
      center && "coop-u-flex-center",
      reverse && "coop-u-flex-row-reverse",
      className,
    ])}
    {...props}
  >
    {children}
  </div>
);

Grid.defaultProps = {
  className: "",
  center: false,
  reverse: false,
  children: null,
};

Grid.propTypes = {
  className: PropTypes.string,
  center: PropTypes.bool,
  reverse: PropTypes.bool,
  children: PropTypes.node,
};

export default Grid;
