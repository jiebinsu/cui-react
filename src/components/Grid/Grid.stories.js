import React from "react";
import Grid from "./index";
import GridItem from "../GridItem";

export default {
  title: "Base/Grid",
  component: Grid,
};

export const Basic = (args) => (
  <Grid {...args}>
    <GridItem size={3}>
      <span
        style={{
          display: "block",
          minHeight: "100px",
          border: "1px solid #6e6e6e",
          background: "#e2e2e2",
        }}
      />
    </GridItem>
    <GridItem size={3}>
      <span
        style={{
          display: "block",
          minHeight: "100px",
          border: "1px solid #6e6e6e",
          background: "#c4c4c4",
        }}
      />
    </GridItem>
  </Grid>
);
Basic.args = {};
