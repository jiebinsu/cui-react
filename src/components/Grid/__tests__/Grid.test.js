import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Grid from "../index";

describe("Grid", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Grid />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<Grid />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow center and reverse attributes", () => {
    const Attributes = () => (
      <>
        <Grid center />
        <Grid reverse />
        <Grid center reverse />
      </>
    );
    const tree = renderer.create(<Attributes />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with children", () => {
    const tree = renderer.create(<Grid>test</Grid>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
