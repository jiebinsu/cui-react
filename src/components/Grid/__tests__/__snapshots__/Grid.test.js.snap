// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should allow center and reverse attributes 1`] = `
Array [
  <div
    className="coop-l-grid coop-u-flex-center"
  />,
  <div
    className="coop-l-grid coop-u-flex-row-reverse"
  />,
  <div
    className="coop-l-grid coop-u-flex-center coop-u-flex-row-reverse"
  />,
]
`;

exports[`Grid should render correctly 1`] = `
<div
  className="coop-l-grid"
/>
`;

exports[`Grid should render with children 1`] = `
<div
  className="coop-l-grid"
>
  test
</div>
`;
